src/Adapter/Shipment/CommandHandler/AddShipmentHandler.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Shipment\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Command\AddShipmentCommand;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\CommandHandler\AddShipmentHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\CannotAddShipmentException;
+use PrestaShopBundle\Entity\Repository\ShipmentRepository;
+use PrestaShopBundle\Entity\Shipment;
+use Throwable;
+
+#[AsCommandHandler]
+class AddShipmentHandler implements AddShipmentHandlerInterface
+{
+    public function __construct(
+        private readonly ShipmentRepository $repository
+    ) {
+    }
+
+    public function handle(AddShipmentCommand $command): int
+    {
+        $shipment = new Shipment();
+        $shipment->setOrderId($command->getOrderId());
+        $shipment->setCarrierId($command->getCarrierId());
+        $shipment->setTrakingNumber($command->getTrackingNumber());
+        $shipment->setAddressId($command->getAddressId());
+        $shipment->setShippingCostTaxExcluded(
+            $command->getShippingCostTaxExcluded()
+        );
+        $shipment->setShippingCostTaxIncluded(
+            $command->getShippingCostTaxIncluded()
+        );
+        foreach ($command->getProducts() as $product) {
+            $shipment->addShipmentProduct($product);
+        }
+
+        if ($command->getPackedAt() !== null) {
+            $shipment->setPackedAt($command->getPackedAt());
+        }
+
+        if ($command->getShippedAt() !== null) {
+            $shipment->setShippedAt($command->getShippedAt());
+        }
+
+        if ($command->getDeliveredAt() !== null) {
+            $shipment->setDeliveredAt($command->getDeliveredAt());
+        }
```

Comment:
For the same reason I'm not sure these dates are relevant in this creation, if the shipment has just been created then these dates are necessarily empty no? Unless you want to create a shipment after it's already been delivered because it didn't exist yet?

But again it's similar to my question above, the workflow of the shipment creation is a bit fuzzy to me 🤔